src/interfaces/filing-interfaces/filing-interface.ts

```diff
@@ -69,6 +69,38 @@ export interface IncorporationFilingIF {
   }
 }
 
+/** Interface for incorporation filing data saved to the Legal API. */
+export interface RegistrationIF {
+  header: {
+    name: string
+    certifiedBy: string
+    date: string
+    effectiveDate?: string // Optional and should be set only for future effective filings
+    filingId?: number // Optional as this is not required when building a filing - causes an error for new filings
+    folioNumber?: string // Optional to the user and only displayed for certain account types
+    isFutureEffective: boolean
+  },
+  business: {
+    legalType: string,
+    identifier: string
+  },
+  registration: {
+    // NB: nameRequest must match schema
+    nameRequest: {
+      legalType: string
+      nrNumber?: string // only set when there is an NR
+      legalName?: string // only set when there is an NR
+    },
+    offices: BaseAddressObjIF | {},
```

Comment:
I'm not exactly sure how an "empty office" is used, but an empty object seems suspicious: it's easier to check for a null office than an empty office object.

I see this isn't the only object declared as possibly empty in this file so 🤷‍♂️

Reply:
hmm yea, in an interface i would expect ONLY the type
Will update this.